python/node2vec_online.py

```diff
@@ -0,0 +1,249 @@
+
+        Example call (with predefined parameters):
+           CALL node2vec_online.set_stream_walk_updater() YIELD *;
+    """
+    global node2vec_context
+
+    if node2vec_context.is_initialized():
+        return mgp.Record(
+            message="StreamWalk updater is already initialized.Call: `CALL node2vec_online.reset() YIELD *` in order to set parameters again. "
+                    "Warning: all embeddings will be lost.")
+
+    node2vec_context.updater = StreamWalkUpdater(
+        half_life=half_life,
+        max_length=max_length,
+        beta=beta,
+        cutoff=cutoff,
+        sampled_walks=sampled_walks,
+        full_walks=full_walks,
+    )
+    return mgp.Record(message="Parameters for StreamWalk updater are set.")
+
+
+@mgp.read_proc
+def set_word2vec_learner(
+        ctx: mgp.ProcCtx,
+        embedding_dimension: int = 128,
+        learning_rate: mgp.Number = 0.01,
+        skip_gram: bool = True,
+        negative_rate: mgp.Number = 10,
+        threads: mgp.Nullable[int] = None,
+) -> mgp.Record(message=str):
+    """
+        This function needs to be called to set parameters.
+
+        If parameters are already set, function doesn't reset parameters.
+
+        :param embedding_dimension number of dimensions of the representation of embedding vector
+        :param learning_rate learning rate
+        :param skip_gram use skip-gram model
+        :param negative_rate negative rate for skip-gram model
+        :param threads maximum number of threads for parallelization
+        :return: empty record
+
+
+       Example call (give all information):
+           CALL node2vec_online.set_gensim_word2vec_learner(128, 0.01, True, 10, 1) YIELD *;
+
+       Example call (with predefined parameters):
+           CALL node2vec_online.set_gensim_word2vec_learner() YIELD *;
+    """
+
+    global node2vec_context
+
+    if node2vec_context.is_initialized():
+        return mgp.Record(
+            message="Word2Vec learner is already initialized. Call: `CALL node2vec_online.reset() YIELD *` in order to set parameters again. "
+                    "Warning: all embeddings will be lost.")
+
+    if threads is None:
+        threads = multiprocessing.cpu_count()
+
+    node2vec_context.learner = GensimWord2Vec(
+        embedding_dimension=embedding_dimension,
+        learning_rate=learning_rate,
+        skip_gram=skip_gram,
+        negative_rate=negative_rate,
+        threads=threads,
+    )
+    return mgp.Record(message="Parameters for Word2Vec learner are set.")
+
+
+@mgp.read_proc
+def reset() -> mgp.Record(message=str):
+    """
+        This function resets parameters and embeddings already learned.
+    """
+
+    global node2vec_context
+
+    node2vec_context.start_time = None
+    node2vec_context.updater = None
+    node2vec_context.learner = None
+    return mgp.Record(message="Updater and learner are ready to be set again.")
```

Comment:
Name `reset` in this case is not doing what is usually intended to, maybe we should come up with the new one

Reply:
maybe `set_clean_state()` I don't have better ideas

Reply:
Let there be a reset, for now, I neither have a better idea